gax/src/main/java/com/google/api/gax/rpc/Watchdog.java

```diff
@@ -138,12 +140,18 @@ public void shutdown() {
 
   @Override
   public boolean isShutdown() {
-    return executor.isShutdown();
+    if (shouldShutdownExecutor) {
```

Comment:
What exactly is the problem we are trying to solve here? Is the problem that Watchdog shutting down executor regardless of the `shouldAutoClose()` flag or Watchdog reporting `isShutdown()` regardless of the `shouldAutoClose()` flag? The implementation seems try to solve the latter.